lib/rules/no-loop-func.js

```diff
@@ -187,7 +187,7 @@ module.exports = {
             if (references.length > 0 &&
                 !references.every(isSafe.bind(null, loopNode))
             ) {
-                context.report({ node, message: "Don't make functions within a loop." });
+                context.report({ node, message: "Don't make functions within a loop that contain unsafe references." });
```

Comment:
If you want to add the variable names to the error message, you could do something like this:

* A few lines up: Instead of checking for `references.every(isSafe.bind(null, loopNode))`, we could use `filter` to get the unsafe references and put them in an array. (Maybe `const unsafeVars = references.filter(r => !isSafe(loopNode, r)).map(r => r.identifier.name)`)
* Then on this line, you could add a message data placeholder: `message: "Don't make functions within a loop that contain unsafe references to variables {{ unsafeVars }}."`
* And you could add a `data` property to the `context.report` object, with something like: `data: { unsafeVars: unsafeVars.join(", ") }`

If that sounds like a massive pain, I could try pushing a change to your branch when I get some free time.

Reply:
I'll give it a go

Reply:
Ok, I've done it. The message now is like this:
Function declared in a loop contains unsafe references to variable(s) 'i', 'j'.

Note: the travis build failure is due to some test timing out, it doesn't appear related to this change.